### PR TITLE
Check order type is set before returning to prevent notice.

### PR DESCRIPTION
**Pull request by vedanshujain**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

When getting order type of a non-order ID, we were emitting a warning. This PR fixes the warning by adding a check to make sure that order type exists before returning.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Create a product, note its ID.
2. Run the code ` Automattic\WooCommerce\Utilities\OrderUtil::is_order( $product_id );`
3. On the trunk, you will see a warning. On this PR, no warning will be emitted.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php [hunk 1]
```diff
@@ -948,7 +948,7 @@ public function get_orders_type( $order_ids ) {
 	 */
 	public function get_order_type( $order_id ) {
 		$type = $this->get_orders_type( array( $order_id ) );
-		return $type[ $order_id ];
+		return $type[ $order_id ] ?? '';
```
Comment (barryhughes) [inline plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:951]: *Not a request for a change,* as such, just an observation: I see a few differences between this and the established CPT implementation.

```
$cpt->get_order_type( $valid_product_id ); # => string(7) "product"
$cot->get_order_type( $valid_product_id ); # => strint(0) ""

$cpt->get_order_type( $invalid_id ); # => bool(false)
$cot->get_order_type( $invalid_id ); # => strint(0) ""
```

I think the approach here is 'more correct' (it correctly satisfies the documented interface, for one thing), and can't immediately think of a case where this difference will lead to issues, but figured I would note it.

**Review by barryhughes — DISMISSED** [review]
Looks good! Approving, but not merging right away so you can review a note I left (I don't think we require a change, but wanted to check-in quickly to err on the safe side).

File: plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php (hunk 1 above)
Comment (vedanshujain) [reply plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:951]: Yeah, I think CPT data store mimics the WP_Post behavior. On further thinking after your comment, I removed the return type (`string?`) from the util method declaration, as it wouldn't have worked correctly with CPT store (and would have raised an error) in https://github.com/woocommerce/woocommerce/pull/35349/commits/acebc8dafe1efce9b6618c8c0e16d47f41cf02af. So probably another review is needed.

Comment (barryhughes) [reply plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php:951]: Good idea.

**Review by barryhughes — APPROVED** [review]
LGTM, thanks!

Blocked from merging because of [this issue](https://github.com/woocommerce/woocommerce/issues/35369), but I pushed a small commit to address that. As soon as checks pass we can merge.

**Comment by barryhughes** [comment]
...Re-running failed checks (seem like random and unrelated fails)...

**Pull request merged**